modules/drivers/sparksql/src/metabase/driver/hive_like.clj

```diff
@@ -158,6 +160,82 @@
     (recur driver hsql-form (* amount 3) :month)
     (hx/+ (hx/->timestamp hsql-form) (hsql/raw (format "(INTERVAL '%d' %s)" (int amount) (name unit))))))
 
+(defmethod sql.qp/->honeysql [:sparksql :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)]
+    (case unit
+      :year
+      (let [positive-diff
+            (fn [a b]
+              (hx/cast
+               :integer
+               (hx/floor (hx// (hx/- (hsql/call :months_between b a)
+                                     (hx/cast :integer (hsql/call :>
+                                                                  (hsql/call :dayofmonth a)
```

Comment:
You call `:dayofmonth` so many times it might be a good idea to abstract it out into a separate function just to make all this stuff a little more concise